### PR TITLE
Mr/procure next

### PR DESCRIPTION
**Pull request by SeduroDotCom**

### CODE REVIEW

File: src/leihs/core/db/type_conversion.clj [hunk 1]
```diff
@@ -39,15 +34,18 @@
...
-  (let [type  (.getType v)
+  (let [type (.getType v)
         value (.getValue v)]
     (if (#{"jsonb" "json"} type)
       (when value
-        (with-meta (<-json value) {:pgtype type}))
+        (let [json-expected (<-json value)]
+          (if (instance? String json-expected)
+            json-expected
+            (with-meta (<-json value) {:pgtype type}))))
```
Comment (SeduroDotCom) [inline src/leihs/core/db/type_conversion.clj:49]: FYI: had to handle ?wrong-json-str? for `class java.lang.String cannot be cast to class clojure.lang.IObj`

https://github.com/leihs/leihs-procure/blob/d0750d12cc2f7e9fde5c98d3189301cb60ac26a6/_tmp/errors/jsonb-string-to-object-cast-exception.txt

1. JSON-CAST-Issue used by procurement_uploads::metadata & ./spec/features/request/admin_crud.feature:37
2. See https://github.com/leihs/leihs-procure/blob/0554fe84fd83702366e9cb3c1211c076340e9f70/server/src/leihs/procurement/graphql.clj#L61-L82  - No clue how to handle non-existing meta-data here

File: src/leihs/core/utils.clj
Comment (nimaai) [inline src/leihs/core/utils.clj:1]: the `my-cast` function and approach should be removed. the following should be done analog to how it is solved in borrow:
* introduce a new type in schema.edn `UUID`
* implement a parser function for this type (string->uuid)

in this way the uuid type cast will be done automatically with added benefit of type validation in the schema itself, etc.

File: src/leihs/core/db/type_conversion.clj (hunk 1 above)
Comment (nimaai) [reply src/leihs/core/db/type_conversion.clj:49]: we most probably need a type cast on the schema.edn level, similar to UUID (see my comment for `utils.clj` and the `my-cast` function)

File: src/leihs/core/auth/session2.clj
Comment (nimaai) [inline src/leihs/core/auth/session2.clj:1]: you can use the original `src/leihs/core/auth/session.clj` from procure. the respective function from this file can continue running through the old `tx` & `jdbc` setup until the whole old tx setup is removed completely from `shared-clj` when `admin` is done. 

then all the concerned `shared-clj` file will be adjusted to `next.jdbc` & `tx-next` etc.

means: you don't need any of your `...2.clj` file versions from `shared-clj`.

**Review by nimaai — CHANGES_REQUESTED** [review]

File: src/leihs/core/auth/core2.clj
Comment (nimaai) [inline src/leihs/core/auth/core2.clj:1]: see https://github.com/leihs/leihs-clj-shared/pull/3#discussion_r1440658672

File: src/leihs/core/auth/shared2.clj
Comment (nimaai) [inline src/leihs/core/auth/shared2.clj:1]: see https://github.com/leihs/leihs-clj-shared/pull/3#discussion_r1440658672

File: src/leihs/core/auth/token2.clj
Comment (nimaai) [inline src/leihs/core/auth/token2.clj:1]: see https://github.com/leihs/leihs-clj-shared/pull/3#discussion_r1440658672

File: src/leihs/core/locale2.clj
Comment (nimaai) [inline src/leihs/core/locale2.clj:1]: see https://github.com/leihs/leihs-clj-shared/pull/3#discussion_r1440658672

File: test/leihs/core/dependencies_test.clj
Comment (nimaai) [inline test/leihs/core/dependencies_test.clj:1]: i am not sure what this file does. i guess it's temporary and can be removed...

File: src/leihs/core/utils.clj
Comment (SeduroDotCom) [reply src/leihs/core/utils.clj:1]: I introduced :ID (:UUID) in schema.edn based on borrow-implementation which reduced the need to use my-cast - but my-cast is still needed for following casts:

https://github.com/leihs/leihs-procure/blob/48d69be04d5e0f1ef1f7d576821b0cdd16d218ff/server/src/leihs/procurement/utils/helpers.clj#L34-L41

DB:
1. procurement_budget_periods (inspection_start_date/end_date timestamp with time zone)

File: src/leihs/core/db/type_conversion.clj (hunk 1 above)
Comment (SeduroDotCom) [reply src/leihs/core/db/type_conversion.clj:49]: FYI: I could remove my-cast and introduced **:ID (UUID) :Int** and **:Timestamptz** in schema.edn

**Pull request closed without merging**